### PR TITLE
Fix(eos_cli_config_gen): Support vars on play via `vars` or `vars_files`

### LINKED ISSUE #2711: Play vars / vars_files are ignored by eos_cli_config_gen
### Issue Summary

When using `vars` or `vars_files` on play-level, the variables are not picked up by `eos_cli_config_gen`. Maybe there is also a problem with `eos_designs` although some vars _did_ work.

### Which component(s) of AVD impacted

eos_cli_config_gen

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

```shell
Add `vars_files` to the playbook:
---
# build.yml

- name: Build Configs
  hosts: DC1_FABRIC
  gather_facts: false
  vars_files:
    - myglobalvars.yml
  tasks:

    - name: Generate AVD Structured Configurations and Fabric Documentation
      ansible.builtin.import_role:
        name: arista.avd.eos_designs

    - name: Generate Device Configurations and Documentation
      ansible.builtin.import_role:
        name: arista.avd.eos_cli_config_gen
```

### Relevant log output

```shell
N/A
```

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Support vars on play via `vars` or `vars_files`

## Related Issue(s)

Fixes #2711

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

Fix variable parsing in `eos_cli_config_gen` to also read variables set on "play".

## How to test

- Added molecule test and tested before and after the change.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Review by carlbuchmann — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/plugins/action/validate_and_template.py [hunk 1]
```diff
@@ -42,8 +42,21 @@ def run(self, tmp=None, task_vars=None):
...
+            if self._templar.is_template(task_vars[var]):
+                # Var contains a jinja2 template.
+                try:
+                    task_vars[var] = self._templar.template(task_vars[var], fail_on_undefined=False)
+                except Exception as e:
+                    raise AnsibleActionFail(f"Exception during templating of task_var '{var}'") from e
+
+        if not isinstance(task_vars, dict):
+            # Corner case for ansible-test where the passed task_vars is a nested chain-map
+            task_vars = dict(task_vars)
```
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/action/validate_and_template.py:59]: Should we not rather pass the correct type from ansible-test?

Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/plugins/action/validate_and_template.py:59]: we don't control that

**Review by gmuloc — APPROVED** [review]
